app/services/images/delete.rb

```diff
@@ -0,0 +1,25 @@
+# app/services/images/delete.rb
+
+module Images
+  class Delete
+    attr_accessor :image_paths
+
+    def initialize(image_paths)
+      @image_paths = image_paths
+    end
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def call
+      uploader = ArticleImageUploader.new
+      image_paths.each do |image_path|
+        next if image_path.blank?
+
+        uploader.retrieve_from_store!(image_path)
+        uploader.remove!
+      end
```

Comment:
If this breaks halfway through (for example, a connection error to S3, on image 4 of 6), what happens when the job retries?

Reply:
@jgaskins the worker has been changed to accept only one image, so it's 100% idemponent. Each worker will work on a single image. See https://github.com/forem/forem/pull/14141/commits/6ac63c899733fb9507e8b60931c9f4a0e6042ae5